src/compiler/parser-proc.bas

```diff
@@ -1197,12 +1228,25 @@ function cProcHeader _
 
 		case else
 			if( is_memberproc ) then
-				if( attrib and FB_SYMBATTRIB_STATIC ) then
-					errReport( FB_ERRMSG_OPERATORCANTBESTATIC, TRUE )
-					attrib and= not FB_SYMBATTRIB_STATIC
+				'' self-op? can't be static
```

Comment:
Above, there is already a check with `astGetOpIsSelf()` (with the TODO) - feels like these two blocks of code should be combined. Probably that TODO is already solved in this block here.

Also, above, the new/delete self-ops are given special treatment - made static automatically. So I'm wondering what happens if you declare them twice, once with `static` and once without. Should be a duplicated definition then I suppose.

Reply:
Yeah, I think I need to rethink this logic of `static` and `astGetOpIsSelf()`

new/delete self-ops will generate duplicate definition if declared twice, with & without static.  Can add a few tests similar to:

```
' TEST_MODE : COMPILE_ONLY_FAIL

type T
	__ as integer
	declare operator new( byval size as uinteger ) as any ptr
	declare static operator new( byval size as uinteger ) as any ptr	
end type
```